ls1intum/Artemis
# Orion: Client Changes in preparation for 1.0.0

### Checklist

### Motivation and Context


The connector between Orion and the client got too big (on the Orion site). We had to split it up into smaller subconnectors, which cover their own set of functionalities. 
The client was not consistent regarding the naming of components/services and properties. We often had "IntelliJ" in these names, although we should stick to a more generic convention.

### Description

- I renamed all remaining occurrences of "intellij..." to "orion..." or "ide..."
- I adapted the client interfaces to fit the new Orion architecture
- There also is a new version checker, that compares the required Orion version with the one used in the actual installed plugin. If the version is not in the allowed range, the user gets automatically routed to an error page, that informs him which version he uses and which is supported.
- The displayed test results in IntelliJ are now also reflect the task hierarchy from the exercise instructions, e.g. if you have: Task 1(Subtask 1, Subtask 2), Task2; then you will have the following test result structure: Task1[with test results for Task 1](Subtask 1[with test results for subtask 1]......

### Steps for Testing

**Outdated version**: [orion_outdated-0.9.0-rc.1.zip](https://github.com/ls1intum/Artemis/files/4091657/orion_outdated-0.9.0-rc.1.zip)
**Working version**: [orion-1.0.0-rc.2.zip](https://github.com/ls1intum/Artemis/files/4144887/orion-1.0.0-rc.2.zip)



1. Log in to Artemis
2. Import an exercise as a student/instructor
3. Run the tests remotely
4. All results should be properly formatted

Test the outdated version checker:
1. Log in to Artemis with an outdated version of Orion
2. You should see the error page informing you about the required version

### Screenshots


Outdated version warning:
![image](https://user-images.githubusercontent.com/27979674/72815972-c43f4400-3c67-11ea-8ddf-89de7a9ba746.png)

Structured test results:
![orion_structured_test_results](https://user-images.githubusercontent.com/27979674/72816047-e0db7c00-3c67-11ea-99cc-efb28608cf1e.gif)
